File: Java/src/main/java/com/gildedrose/model/AgedBrie.java
package com.gildedrose.model;

import static com.gildedrose.Constants.MAX_QUALITY;

public class AgedBrie extends Goods {

    public static final String AGED_BRIE = "Aged Brie";

    public AgedBrie(int sellIn, int quality) {
        super(AGED_BRIE, sellIn, quality);
    }

    @Override
    public void updateQuality() {

        sellInPasses();
        upgradeQuality();
        if (sellIn < 0) {
            upgradeQuality();
        }

    }

    private void upgradeQuality() {
        quality = Math.min(quality + 1, MAX_QUALITY);
    }

}

File: python/src/aged_brie.py
from python.components.gilded_rose import GildedRose, Item


class AgedBrieLogic(GildedRose):

    def __init__(self, item: Item):
        super().__init__(item)

    def update_quality(self):
        if self.quality < 50:
            self.quality += 1
        self.sell_in -= 1
        if self.sell_in < 0 and self.quality < 50:
            self.quality += 1

        return self.quality, self.sell_in